src/runner/fixture-hook-controller.ts

```diff
@@ -103,15 +126,9 @@ export default class FixtureHookController {
         if (item) {
             item.pendingTestRunCount--;
 
-            if (item.pendingTestRunCount === 0 && fixture.afterFn) {
-                testRun.phase = TEST_RUN_PHASE.inFixtureAfterHook;
-
-                try {
-                    await fixture.afterFn(item.fixtureCtx);
-                }
-                catch (err) {
-                    testRun.addError(processTestFnError(err));
-                }
+            if (item.pendingTestRunCount === 0) {
+                await this._runFixtureAfterHook(item, fixture.globalAfterFn, testRun);
+                await this._runFixtureAfterHook(item, fixture.afterFn, testRun);
```

Comment:
There is inconsistency between runFixtureAfterHook and runFixtureBeforeHook.
The `runFixtureBefore` hook can be executed once if the first hook failed. However, the `runFixtureAfterHook` will be executed twice in any cases

Reply:
I changed the order of the hooks. Prohibiting the next 'after' hook is maybe wrong because both hooks may be independent. Also, the second hook may be finishing an important process without one next fixture won't start.